### PR TITLE
ENH: stats.tvar/tstd/tsem: add array API support

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Toward gh-20544

#### What does this implement/fix?
Adds array API support to `tvar`, `tstd`, and `tsem`.

### CODE REVIEW

File: scipy/stats/tests/test_stats.py [hunk 1]
```diff
@@ -257,15 +248,14 @@ def test_tax_scalar_and_nanpolicy(self):
...
                 stats.tmax(x, nan_policy='foobar')
 
-    def test_tsem(self):
-        y = stats.tsem(X, limits=(3, 8), inclusive=(False, True))
-        y_ref = np.array([4, 5, 6, 7, 8])
-        assert_approx_equal(y, y_ref.std(ddof=1) / np.sqrt(y_ref.size),
-                            significant=self.dprec)
+    def test_tsem(self, xp):
+        x = xp.asarray(X)
+        xp_test = array_namespace(x)  # need array-api-compat var for `correction`
```
Comment (lucascolley) [inline scipy/stats/tests/test_stats.py:253]: `std` rather than `var`? here and above

**Review by lucascolley — COMMENTED** [review]
can take a closer look later but all looks good from a scan

File: scipy/stats/tests/test_stats.py (hunk 1 above)
Comment (mdhaber) [reply scipy/stats/tests/test_stats.py:253]: Sure, copy-paste error. Fixed.

**Comment by mdhaber** [comment]
I will push again once I fix the torch issues and confirm locally.

File: scipy/stats/tests/test_stats.py [hunk 2]
```diff
@@ -73,22 +73,21 @@
...
-    @array_api_compatible
-    @skip_xp_backends('array_api_strict', 'jax.numpy',
-                      reasons=["`array_api_strict.where` `fillvalue` doesn't "
-                               "accept Python floats. See data-apis/array-api#807.",
-                               "JAX doesn't allow item assignment, and this  "
-                               "function uses _lazywhere."])
-    @pytest.mark.usefixtures("skip_xp_backends")
     def test_tmean(self, xp):
-        x = xp.asarray(X)
+        x = xp.asarray(X.tolist())  # use default dtype of xp
```
Comment (j-bowhay) [inline scipy/stats/tests/test_stats.py:87]: This is a neat trick I hadn't thought of doing this

**Review by j-bowhay — APPROVED** [review]
Thanks LGTM!

The comment on line 63 of `test_stats.py` is somewhat amusing!

File: scipy/stats/tests/test_stats.py (hunk 2 above)
Comment (lucascolley) [reply scipy/stats/tests/test_stats.py:87]: How portable is this out of interest?

Comment (mdhaber) [reply scipy/stats/tests/test_stats.py:87]: `tolist` says it converts NumPy scalars to Python scalars, and it sounds like a Python `float` is [all but guaranteed](https://stackoverflow.com/questions/70184494/on-what-systems-does-python-not-use-ieee-754-double-precision-floats) to be an IEEE 754 double.

**Pull request merged**